src/Data/Aeson/TH.hs

```diff
@@ -327,8 +327,8 @@ consToValue :: ToJSONFun
             -- ^ Constructors for which to generate JSON generating code.
             -> Q Exp
 
-consToValue _ _ _ _ [] = error $ "Data.Aeson.TH.consToValue: "
-                             ++ "Not a single constructor given!"
+consToValue _ _ _ _ [] =
+    [| \x -> x `seq` error "case: V1" |]
```

Comment:
use `EmptyCase`.

Reply:
Done here as well, however, I moved the expression into a separate function, otherwise the empty-case expression in the splice will require enabling `EmptyCase` at the use site too